app/models/search/collection_indexer.rb

```diff
@@ -0,0 +1,115 @@
+class CollectionIndexer < Indexer
+
+  def self.klass
+    "Collection"
+  end
+
+  def self.mapping
+    {
+      "collection" => {
+        properties: {
+          title: {
+            type: "text",
+            analyzer: "collection_title_analyzer",
+            fields: {
+              exact: {
+                type:     "text",
+                analyzer: "exact_collection_analyzer"
+              },
+              keyword: {
+                type: "keyword",
+                normalizer: "keyword_lowercase"
+              }
+            }
+          },
+          challenge_type: {            
+            type: "keyword",
+            null_value: "NULL" 
+          },
+          name: { type: "text", analyzer: "simple" },
+          description: { type: "text", analyzer: "simple" },
+          collection_type: { type: "keyword" },
+          created_at: { type: "date" }
+        }
+
+      }
+    }
+  end
+
+  def self.settings
+    {
+      analysis: {
+        analyzer: {
+          collection_title_analyzer: {
+            type: "custom",
+            tokenizer: "standard",
+            filter: [
+              "lowercase"
+            ]
+          },
+          exact_collection_analyzer: {
+            type: "custom",
+            tokenizer: "keyword",
+            filter: [
+              "lowercase"
+            ]
+          }
+        },
+        normalizer: {
+          keyword_lowercase: {
+            type: "custom",
+            filter: ["lowercase"]
+          }
+        }
+      }
+    }
+  end
+
+  def document(object)
+    object.as_json(
+      root: false,
+      only: [
+        :id, :name, :title, :description, :parent_id, :collection_type, :created_at
+      ]
+    ).merge(
+      closed: object.closed?,
+      unrevealed: object.unrevealed?,
+      anonymous: object.anonymous?,
+      owner_ids: object.all_owners.pluck(:user_id),
+      moderated: object.moderated?,
+      moderator_ids: object.all_moderators.pluck(:user_id),
+      challenge_type: object.challenge_type,
+      signup_open: object.challenge&.signup_open,
+      signups_open_at: object.challenge&.signups_open_at,
+      signups_close_at: object.challenge&.signups_close_at,
+      assignments_due_at: object.challenge&.assignments_due_at,
+      works_reveal_at: object.challenge&.works_reveal_at,
+      authors_reveal_at: object.challenge&.authors_reveal_at,
+      general_fandom_ids: get_fandom_ids(object),
+      public_fandom_ids: get_fandom_ids(object, true),
+      general_fandoms_count: get_fandom_ids(object).count,
```

Comment:
I don't think using `get_fandom_ids(object).count` will work here, since [the issue](https://otwarchive.atlassian.net/browse/AO3-6026)/SOW notes:

>general_fandoms_count is the total number of fandoms on all public and restricted works approved for inclusion in the collection or any of its subcollections. Unlike the general_fandom_ids, this field counts only direct filter_taggings. It will be used in blurbs to display the fandom count to logged-in users.

For example...

Let's say we have the canonical fandom tag "Marvel Cinematic Universe" and it's the metatag of the canonical fandom "Captain America Movies."

If I post a work with the "Captain America Movies" fandom tag, it will have a direct [filter_tagging](https://github.com/otwcode/otwarchive/wiki/Guide:-Tag-Code#filtertaggings) for that, but _also_ an inherited filter_tagging for "Marvel Cinematic Universe." That way, if someone goes to the work page for the "Marvel Cinematic Universe" tag, they will see my work about Captain America listed there.

However, if I make the collection "Best Captain America Works Ever" and add my work to it, when I go to my collection's Fandoms page (located at a path like `/collections/best_captain_america_works_ever/fandoms`), the only fandom listed there will be "Captain America Movies" -- the [list on that page is drawn from _direct_ filter taggings](https://github.com/otwcode/otwarchive/blob/cbfa1db7e6dd16d9ca2807d63bed7d736aa88f5e/app/controllers/fandoms_controller.rb#L12-L13).

So the idea is general_fandoms_count and public_fandoms_count should align with the number of fandoms you'll see when you go to the collection's Fandoms page. Otherwise, we'll end up in a situation where someone sees "Fandoms: 2" on the blurb for the "Best Captain America Works Ever" collection, but when they go to /collections/best_captain_america_works_ever/fandoms, they'll only see one fandom, leading them to believe there's a) content they can't see or b) a bug.

Reply:
Thanks for this feedback. I was worried that this data wouldn't be quite correct. I had a bit of a hard time wrapping my head around the structure of the data, by just looking at the methods on the collection model.
Do you have any tips for how to compile the data correctly for these methods (and the comment above)?

Reply:
Before I try to be helpful, two notes:
1. Someone else might come along and say doing it this way will make the servers cry and there's a better way to do it (I'm guessing a complex join), in which case you should listen to them.
2. Someone else might come along and say this is a terrible abuse of  these methods, in which case you should also listen to them.

That said, [Filterable](https://github.com/otwcode/otwarchive/blob/master/app/models/concerns/filterable.rb) has some methods that might be useful. You can use them on both works and bookmarkables.

Here's a messy example I knocked together on the console for calculating a collection's `general_fandom_ids`, which includes both inherited and direct Fandom filter taggings:
```ruby
bookmarkable_fandom_ids = []
work_fandom_ids = []

approved_bookmarks.map do |bookmark|
  bookmarkable_fandom_ids << bookmark.bookmarkable.filters.where(type: "Fandom").pluck(:filter_id)
end

approved_works.map do |work|
  work_fandom_ids << work.filters.where(type: "Fandom").pluck(:filter_id)
end

general_fandom_ids = [work_fandom_ids + bookmarkable_fandom_ids].flatten.uniq
```

For the `general_fandoms_count`, which only needs direct filter taggings, we can replace `filters` with [`direct_filters`](https://github.com/otwcode/otwarchive/blob/3785907b55d3d844e4d73deb54f0a8f8c3d9a44b/app/models/concerns/filterable.rb#L17-L19) and then count the results.... or [`fandom_ids`](https://github.com/otwcode/otwarchive/blob/3785907b55d3d844e4d73deb54f0a8f8c3d9a44b/app/models/concerns/filterable.rb#L91-L93) should also work and be much shorter, e.g. `work.fandom_ids`.

It's probably also worth noting that just [`collection.fandoms`](https://github.com/otwcode/otwarchive/blob/3785907b55d3d844e4d73deb54f0a8f8c3d9a44b/app/models/collection.rb#L64) will get you the direct filter tagging Fandoms for all approved works in the collection.

And one last note before you get too involved in working on the counts: I just noticed something in our SOW/issue _might_ be wrong. Currently, we say that the counts should include fandoms listed on bookmarkables. However, right now, the Fandoms page for a collection only includes fandoms used on _works_. I'm going to check  with everyone else, but I'm thinking we might want to stick with the works-only approach for counts, or else we're going to have people telling us, "The blurb said the collection had 10 fandoms, but the sidebar only says 2 and there are only 2 on the Fandoms page."

Reply:
This is all really helpful. So it looks like this _could_ be the answer, but I should hold off on making changes until you check in with the other devs?

Reply:
Yeah, I think we might end up scrapping this fandoms thing, so please hold off!

Reply:
Okay, hi, I have returned! 

We had a talk about just how much reindexing we'd have to do, and we've decided we no longer want to deal with `general_fandom_ids`, `public_fandom_ids`, `general_fandoms_count`, or `public_fandoms_count`. 

What we'd like to do instead is make the collections themselves taggable and filterable. We're still working on exactly what that will entail, but once we figure that out, we'll give you a detailed rundown and you can give us an updated estimate and let us know when you'll have time to work on it. (Assuming you want to, of course!) 

I think we can have that info for you within the next 2.5 weeks (I was going to say 2, but then I realized that would land us right on the U.S. Thanksgiving holiday).

For now, though, we're definitely sure you can remove 
* these fields, as well as the `get_fandom_ids` method and any code for reindexing based on changes to the collection's works' or bookmarks' fandoms
* the corresponding "Filter by fandom" section of `app/views/collections/_filters.html.erb`
* [the fandom count on `app/views/collections/_collection_blurb.html.erb`](https://github.com/otwcode/otwarchive/blob/b5bf2e7a5cd362d1e350f2be087e8049721a2484/app/views/collections/_collection_blurb.html.erb#L46-L49)